homeassistant/components/beewi_smartclim/sensor.py

```diff
@@ -0,0 +1,187 @@
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_MONITORED_CONDITIONS, default=list(SENSOR_TYPES)): vol.All(
+            cv.ensure_list, [vol.In(SENSOR_TYPES)]
+        ),
+    }
+)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the beewi_smartclim platform."""
+    _LOGGER.info("BeeWi SmartClim has been loaded")
+
+    mac = config.get(CONF_MAC)
+
+    poller = BeewiSmartClimPoller(mac)
+
+    sensors = []
+
+    for parameter in config[CONF_MONITORED_CONDITIONS]:
+        device = SENSOR_TYPES[parameter][0]
+        name = SENSOR_TYPES[parameter][1]
+        unit = SENSOR_TYPES[parameter][2]
+
+        prefix = config.get(CONF_NAME)
+        if prefix:
+            name = "{} {}".format(prefix, name)
+
+        sensors.append(BeewiSmartclimSensor(poller, name, mac, device, unit))
+
+    add_entities(sensors)
+
+
+class BeewiSmartclimSensor(Entity):
+    """Representation of a Sensor."""
+
+    def __init__(self, poller, name, mac, device, unit):
+        """Initialize the sensor."""
+        self._poller = poller
+        self._name = name
+        self._mac = mac
+        self._device = device
+        self._unit = unit
+        self._state = None
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the sensor. Convert Celsius to Farhenheit if needed."""
+        if self._device == DEVICE_CLASS_TEMPERATURE and self._unit == TEMP_FAHRENHEIT:
+            return (self._state * 1.8) + 32
+        return self._state
+
+    @property
+    def device_class(self):
+        """Device class of this entity."""
+        return self._device
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return self._unit
+
+    def update(self):
+        """Fetch new state data from the poller."""
+        self._state = self._poller.get_value(self._device)
+
+
+class BeewiSmartClimPoller:
+    """This class will interact with the sensor and aggregates all data."""
+
+    def __init__(self, mac):
+        """Initialize the Poller."""
+        try:
+            from btlewrap import BluepyBackend
+
+            backend = BluepyBackend
+        except ImportError:
+            from btlewrap import GatttoolBackend
+
+            backend = GatttoolBackend
+
+        self._backend = backend
+        self._mac = mac
+        self._temp = None
+        self._humidity = None
+        self._battery = None
+        self._last_update = None
+
+        _LOGGER.debug("MiTempBtSensorPoller initiated with backend %s", self._backend)
+
+    def get_value(self, device):
+        """
+        Return the value from the cached data.
+
+        In the case it's outdated, it will call the update endpoint to refresh those values.
+        """
+        if (self._last_update is None) or (
```

Comment:
We can use the `Throttle` util here instead. Decorate the `update_data` method with a throttle instance.